lib/features/steps/error_reporting_steps.rb

```diff
@@ -23,6 +23,96 @@
   }
 end
 
+# Verifies that an event is correct for an unhandled error
+# This checks various elements of the payload matching an unhandled error including:
+#    The unhandled flag
+#    Any attached session information
+#    Severity
+#
+# @param event [Int] The event to verify
+Then("event {int} is unhandled") do |event|
+  if read_key_path(Server.current_request[:body], "events.#{event}.session").nil?
+    steps %Q{
+      Then the payload field "events.#{event}.unhandled" is true
+      And the payload field "events.#{event}.severity" equals "error"
+    }
+  else
+    steps %Q{
+      Then the payload field "events.#{event}.unhandled" is true
+      And the payload field "events.#{event}.severity" equals "error"
+      And the payload field "events.#{event}.session.events.unhandled" is greater than 0
+    }
+  end
+end
+
+# Verifies that an event is correct for an unhandled error
+# This checks various elements of the payload matching an unhandled error including:
+#    The unhandled flag
+#    Any attached session information
+#    Severity
+#
+# @param event [Int] The event to verify
+# @param severity [String] An expected severity different to the default "error"
+Then("event {int} is unhandled with the severity {string}") do |event, severity|
+  if read_key_path(Server.current_request[:body], "events.#{event}.session").nil?
```

Comment:
It seems like these assertions are shared between all the added steps. Creating some private function and calling it from within each step would make it easier to make a change in one place, e.g.:

```
verify_error(boolean unhandled, string severity)
```